ruby/red-arrow/test/test-table.rb

```diff
@@ -147,6 +147,96 @@ def setup
       TABLE
     end
 
+    test("{key: Number}") do
+      assert_equal(<<-TABLE, @table.slice(count: 16).to_s)
+	count	visible
+0	   16	true   
+      TABLE
+    end
+
+    test("{key: String}") do
+      name_array = Arrow::StringArray.new(["a", "b", "c", "d", "e", "f", "g", "h"])
+      table_with_name = @table.merge(:name => name_array)
+      assert_equal(<<-TABLE, table_with_name.slice(name: 'e').to_s)
+	count	visible	name
+0	   16	true   	e   
+      TABLE
+    end
+
+    test("{key: TrueClass}") do
```

Comment:
We can simplify this because `true` is the only one instance of `TrueClass`:

```suggestion
    test("{key: true}") do
```